frogmouth/utility/type_tests.py

```diff
@@ -34,8 +37,20 @@ def _(resource: str) -> bool:
 
 
 @maybe_markdown.register
-def _(resource: URL) -> bool:
-    return maybe_markdown(resource.path)
+async def _(resource: URL) -> bool:
```

Comment:
While I can see why this would be desirable (even if it's somewhat niche), I'm not sure this is the way go to about it. This suddenly makes `maybe_markdown` into a single dispatch method that is non-async in some cases but async in another (pyright, for example, quite right complains about this). Also, this suddenly turns the check for a possible Markdown file on the end of a URL into a much more expensive test.

I think there's some merit in handling this case, but I don't think this is the way to go about this.